sensory_jamming.py

```diff
@@ -0,0 +1,342 @@
+                    # {bat:{time of calling:[x,y,propdist]}}
+                
+            else:
+                self.callshistory[int(self.timestep)] = 0
+                    # add a "no new call" (accounted for with 0)
+
+
+            if self.ID in env.callsources.keys():
+                # if the agent has ever called before: 
+        
+                for calltime in env.callsources[int(self.ID)].keys():
+                    # for each time step at which a call was emitted:
+                    self.Data_storage(env.callsources, calltime)
+                        # erase calls that are too old to be heared anymore
+                    if calltime in env.callsources[int(self.ID)].keys():
+                        # if the call is still in the dictionary:
+                        self.Propagation(env.callsources, calltime)
+                            # update its propagation distance accordingly
+
+                    
+        def Hearing(self):
+            
+            for identity in env.all_ID:
+                # for each agent in the simulation:
+                
+                if identity in env.callsources.keys():
+                    # if the agent has ever called before: 
+                    
+                    for calltime in env.callsources[int(identity)].keys():
+                        # for each time step at which the agent previously called:
+                        self.Hearing_test(identity, calltime, self.hearhistory)
+                        # identify and record calls that can be heard by focal agent
+    
+        def Boundaries(self, coord, coordbound):
+        
+            if coord > 0 and coord < coordbound:
+                # if the coordinate is within boundaries:
+                return float(coord)
+                    # return the original coordinate.
+            else:
+                return float(0)
+                    # reset the coordinate to the beginning of the space frame.
+        
+        def Data_storage(self, dict1, tcall):
+            self.timestore = self.timestep - tcall
+                # Time (in time steps) for which the call has been stored into 
+                # the dictionary.
+            
+            if self.timestore > self.max_timestore/self.d_t:
+                # if the storing time is longer than it should: 
+                dict1[int(self.ID)].pop(tcall, None)
+                    # erase it from the memory / dictionary
+        
+            return dict1
+
+        def Propagation(self, dict1, tmstp):
+            self.propdist = float(self.speedsound * self.timestore * env.d_t)
+                # propagation distance at timestep according to the time when 
+                # the call was emitted and the speed of sound.
+            dict1[int(self.ID)][int(tmstp)]['propdist'] = self.propdist
+                # update the propagation distance in dict1
+        
+            return dict1
+    
+        def In_ring(self, center_x, center_y, radius, x, y):
```

Comment:
could you change the name of the center_x to reflect the fact that you're referring to the call: eg. callcentre_x & callcentre_y - and also change the 'x' and 'y' to - agent_x and agent_y